arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -226,24 +217,45 @@ static void partial_fold(const size_t len, __m128i *xmm_crc0, __m128i *xmm_crc1,
     *xmm_crc3 = _mm_castps_si128(ps_res);
 }
 
-Z_INTERNAL void crc_fold_copy(unsigned int crc0[4 * 5], unsigned char *dst, const unsigned char *src, long len) {
+static inline void crc32_fold_load(__m128i *fold, __m128i *fold0, __m128i *fold1, __m128i *fold2, __m128i *fold3) {
+    *fold0 = _mm_load_si128(fold + 0);
+    *fold1 = _mm_load_si128(fold + 1);
+    *fold2 = _mm_load_si128(fold + 2);
+    *fold3 = _mm_load_si128(fold + 3);
+}
+
+static inline void crc32_fold_save(__m128i *fold, __m128i fold0, __m128i fold1, __m128i fold2, __m128i fold3) {
+    _mm_storeu_si128(fold + 0, fold0);
+    _mm_storeu_si128(fold + 1, fold1);
+    _mm_storeu_si128(fold + 2, fold2);
+    _mm_storeu_si128(fold + 3, fold3);
```

Comment:
_mm_store_si128

Reply:
`fold` can be unaligned. 

![image](https://user-images.githubusercontent.com/1364432/124365985-f137aa80-dc00-11eb-930b-fca97a9d279e.png)

But `src` appears to always be aligned.

![image](https://user-images.githubusercontent.com/1364432/124366001-090f2e80-dc01-11eb-91cc-fccc70959dab.png)

Reply:
Shrugs... but maybe have two versions, one aligned and one unaligned... On some older processors, unaligned versions of intrinsics are slower because optimizer can't deduct alignment of the variables that are not allocated from stack.

Reply:
I'm ok with the way it is for now.

Reply:
Basically currently the code is only used in two places, so in my opinion, there is equal amount of pros and cons having either shared version or both having own separate version. 

It doesn't really add a lot more code length in compiled binary than the cleanup already does as the code gets inlined anyways and in source form it's only few extra lines (and makes the code more readable).